src/form.js

```diff
@@ -40,38 +40,61 @@ export function setFormForElement(element, form) {
 
 /**
  * Returns form data in the passed-in form as an object.
+ * Includes focused submit buttons.
  * @param {!HTMLFormElement} form
  * @return {!JsonObject}
  */
 export function getFormAsObject(form) {
+  const {
+    elements,
+    ownerDocument,
+  } = form;
   const data = /** @type {!JsonObject} */ ({});
-  const inputs = form.elements;
-  const submittableTagsRegex = /^(?:input|select|textarea)$/i;
+  const submittableTagsRegex = /^(?:input|select|textarea|button)$/i;
   const unsubmittableTypesRegex = /^(?:button|image|file|reset)$/i;
   const checkableType = /^(?:checkbox|radio)$/i;
-  for (let i = 0; i < inputs.length; i++) {
-    const input = inputs[i];
-    const {name} = input;
+  for (let i = 0; i < elements.length; i++) {
+    const input = elements[i];
+    const {
+      checked,
+      name,
+      multiple,
+      options,
+      tagName,
+      type,
+      value,
+    } = input;
     if (!name || isDisabled(input) ||
-        !submittableTagsRegex.test(input.tagName) ||
-        unsubmittableTypesRegex.test(input.type) ||
-        (checkableType.test(input.type) && !input.checked)) {
+        !submittableTagsRegex.test(tagName) ||
+        unsubmittableTypesRegex.test(type) ||
+        (checkableType.test(type) && !checked)) {
       continue;
     }
 
     if (data[name] === undefined) {
       data[name] = [];
     }
 
-    if (input.multiple) {
-      iterateCursor(input.options, option => {
+    if (multiple) {
+      iterateCursor(options, option => {
         if (option.selected) {
           data[name].push(option.value);
         }
       });
-    } else {
-      data[name].push(input.value);
+      continue;
     }
+
+    // Browsers vary on the conditions that focus the form submit button.
```

Comment:
add maybe add "In a form, the button's value is only submitted if the button itself was used to submit the form."